Sources/SwiftParser/Declarations.swift

```diff
@@ -1318,32 +1317,18 @@ extension Parser {
   public mutating func parseFunctionSignature() -> RawFunctionSignatureSyntax {
     let input = self.parseParameterClause(for: .functionParameters)
 
-    let async: RawTokenSyntax?
-    if let asyncTok = self.consume(if: .keyword(.async)) {
-      async = asyncTok
-    } else if let reasync = self.consume(if: .keyword(.reasync)) {
-      async = reasync
-    } else {
-      async = nil
-    }
-
-    var throwsKeyword = self.consume(ifAny: [.keyword(.throws), .keyword(.rethrows)])
+    var effectSpecifiers = self.parseDeclEffectSpecifiers()
 
     let output: RawReturnClauseSyntax?
     if self.at(.arrow) {
```

Comment:
I imagine we wouldn't be recovering well from random junk before the arrow right now? Could this use `canRecoverTo` instead?